src/db.rs

```diff
@@ -381,10 +400,18 @@ impl Db {
             inner.spawn_memtable_flush_task(manifest, memtable_flush_rx, &tokio_handle);
         let mut compactor = None;
         if let Some(compactor_options) = &inner.options.compactor_options {
+            // not to pollute the cache during compaction
+            let uncached_table_store = Arc::new(TableStore::new_with_fp_registry(
```

Comment:
I see from the linked comment that this is so that we don't cache the files read by compactor.
1. While the files that compactor reads wouldn't need to be cached, compactors can surely benefit from reading from the cache. In the case where compactor runs in the same node, L0 files wouldn't need to be downloaded from object store again.
2. Files written by compactor are likely to be used more frequently, so there is indeed benefit to cache them.

This goes in the direction of passing caching hints.

From the comment again, I see that we want to keep this a non-invasive incremental change. With that in mind, keeping it as is should be fine.
Sharing above for discussion for future PRs.

Reply:
this makes sense, i guess we can add a flag to the cached object store to allow it `readonly`, and never write files.

or a standalone option for compactor to select enabling read or writing object store caches as user wishes.

Reply:
https://github.com/slatedb/slatedb/issues/196 - This issue tracks it.